pkg/apis/coordination/validation/validation.go

```diff
@@ -17,21 +17,20 @@ limitations under the License.
 package validation
 
 import (
-	"k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/kubernetes/pkg/apis/coordination"
 )
 
 // ValidateLease validates a Lease.
 func ValidateLease(lease *coordination.Lease) field.ErrorList {
-	allErrs := validation.ValidateObjectMeta(&lease.ObjectMeta, true, validation.NameIsDNSSubdomain, field.NewPath("metadata"))
```

Comment:
The `ValidateObjectMetaAccessor` and `ValidateObjectMeta`functions takes an argument for `nameValidationFn`.

The name validation check done in registry just checks if the name can be safely encoded as a path segment

https://github.com/kubernetes/kubernetes/blob/f15fe9338484ccfe12c530d1acba7d44ae54e8e9/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L112

https://github.com/kubernetes/kubernetes/blob/f15fe9338484ccfe12c530d1acba7d44ae54e8e9/staging/src/k8s.io/apimachinery/pkg/api/validation/path/name.go#L62

For those with custom name validations, we will need to preserve the call. But most of the custom ones just have `validation.NameIsDNSSubdomain` as the check. Resources which have extra checks have this + other restrictions. Can we add this into the name validation function in generic registry?